flexget/plugins/input/ftp_list.py

```diff
@@ -143,8 +148,7 @@ def _handle_path(self, entries, ftp, baseurl, path='', mlst_supported=False, fil
                                   recursive, get_size, encoding)
 
             if not files_only or mlst.get('type') == 'file':
-                url = baseurl + path + '/' + p
-                url = url.replace(' ', '%20')
+                url = baseurl + urllib.quote_plus(path) + '/' + urllib.quote_plus(p)
```

Comment:
OK, the fix was actually a lot easier. I'm just going to use quote(...) instead of quote_plus(...). I also tested the generated URL in Firefox and it works even when I use all sorts of spaces and weird characters in the URL. However, I uncovered an issue with ftp_download in the process. It's not downloading the files to the same folder hierarchy on the local disk. The issue appears to have always been there though? I'm going to try and track it down and fix it.